plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -130,6 +229,127 @@ class bp_connection_manager {
              established_client_connection(new_connection) && num_established_clients() > self()->connections.get_max_client_count();
    }
 
+   // thread safe
+   // removes invalid entries from msg
+   bool validate_gossip_bp_peers_message( gossip_bp_peers_message& msg ) {
+      if (msg.peers.empty())
+         return false;
+      if (msg.peers.size() != 1 || !msg.peers[0].server_address.empty()) { // initial case, no server_addresses to validate
```

Comment:
I was a little misled because I would expect that the comment *after* the `{` describes the `then` state. May I suggest something like:
```suggestion
      bool initial_msg = msg.peers.size() == 1 && msg.peers[0].server_address.empty());
      if (!initial_msg) { 
```